pkg/controller/sparkapplication/controller_test.go

```diff
@@ -287,10 +297,17 @@ func TestOnDelete(t *testing.T) {
 }
 
 func TestProcessSingleDriverStateUpdate(t *testing.T) {
+	type metrics struct {
```

Comment:
Looks like the same type is used and defined in `TestProcessSingleAppStateUpdate` below, and a similar type is defined in `TestProcessSingleExecutorStateUpdate ` for executors. I think we can define a single type at the package level for this and use it for all three tests.

Reply:
done